### ISSUE TITLE
quickRequest: Unexpected behaviour dependent on method order

### ISSUE THREAD

**Issue by nemoo** [softwaremill/sttp#2144]
Version: "com.softwaremill.sttp.client4" %% "core" % "4.0.0-M12"
### Expected behaviour:
Changing the order of methods used with a declarative quickRequest should not alter the behaviour of the request.

### Actual behaviour:
Original oder:
```scala
import sttp.client4.quick._

    val response = quickRequest
      .post(url)
      .headers(
        Map(
          "Accept" -> "application/xml",
          "Content-Type" -> "application/json",
          "X-Rundeck-Auth-Token" -> authToken
        )
      )
      .body(payload.render())
      .readTimeout(Duration(10, SECONDS))
      .response(asStringAlways)
      .send()
```
Changing the order of the headers and the body method changes the behaviour:
```scala
import sttp.client4.quick._

    val response = quickRequest
      .post(url)
      .body(payload.render())
      .headers(
        Map(
          "Accept" -> "application/xml",
          "Content-Type" -> "application/json",
          "X-Rundeck-Auth-Token" -> authToken
        )
      )
      .readTimeout(Duration(10, SECONDS))
      .response(asStringAlways)
      .send()
```
As a new user to sttp I find this super confusing

**Comment by adamw** [comment]
I'd suspect content-types, but these should be identical. What's the return type of `payload.render()`? Can you dump `request.headers` to see where they diverge?

**Comment by nemoo** [comment]
Yes, it produces differant (and in the second case even duplicate within the same request?) Content-Types. Here is a dump of the headers:
```
Vector(Accept-Encoding: gzip, deflate, Content-Type: text/plain; charset=utf-8, Content-Length: 43, Accept: application/xml, Content-Type: application/json, X-Rundeck-Auth-Token: XXXX)
Vector(Accept-Encoding: gzip, deflate, Accept: application/xml, Content-Type: application/json, X-Rundeck-Auth-Token: XXXX, Content-Length: 43)
```

**Comment by adamw** [comment]
The problem that you encountered comes from the fact that `.headers` **adds** the given headers to the headers list; while `.body(...)` sets the content type if one is not defined before. 

The confusion comes probably from the fact that such an approach might be too simplistic: it treats all headers additively (however, there is also a variant to replace a single header). What we could do instead, is have some "smart" logic of replacing headers which are known to be "singular", e.g. it doesn't make sense to have two `Content-Type` headers - as exemplified above.

**Comment by nemoo** [comment]
>  What we could do instead, is have some "smart" logic of replacing headers which are known to be "singular", e.g. it doesn't make  sense to have two Content-Type headers - as exemplified above.

Sounds great. For me as a user it is important that the headers that I set via the .headers method are actually used like defined in the method call. When I see a .headers method in a scala api, I expect it to declaratively set it, not to "add" it. The proposed "smart" logic seems to be able to achieve that.
